functions/debounce.mjs

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable func-names */
+
+/**
+ * debounce :: a -> Time -> a
```

Comment:
Haha, I don't know why I started this. 🐢 

This would actually be `debounce :: (a -> b) -> Number -> (a -> )`.
Actually, whilst writing the signature I notice the inner function not having a return value. I wonder whether we should use promises here to avoid side effects. 

But anyways, since we've been debouncing like this for years, let's merge it. 👍